lib/charon/internal/crypto.ex

```diff
@@ -163,10 +163,49 @@ defmodule Charon.Internal.Crypto do
 
   def verify_hmac(_, _), do: {:error, :malformed_input}
 
+  @doc """
+  Generate a random string containing `digit_count` base10 characters.
+  """
+  @spec random_digits(pos_integer) :: binary
+  def random_digits(digit_count) do
+    random_digits_integer(digit_count)
+    |> Integer.to_string()
+    |> String.pad_leading(digit_count, "0")
+  end
+
+  @doc """
+  Generate a random number with at most `digit_count` digits.
+  """
+  @spec random_digits_integer(pos_integer) :: number
+  def random_digits_integer(digit_count) when digit_count > 0 do
+    boundary = Integer.pow(10, digit_count)
+
+    fn -> :crypto.strong_rand_bytes(5) end
+    |> Stream.repeatedly()
+    |> Enum.reduce_while({_count = 0, _result = 0}, fn <<int1::40>>, acc ->
+      acc
+      |> maybe_add_twelve_digits(int1)
```

Comment:
This helper function, now that you call it only once, can optionally be integrated here.

Reply:
Done in https://github.com/weareyipyip/charon/pull/63/commits/723a0f8782295edc372b7d6dc35ed5fbd618d008